client/reader/sidebar/index.jsx

```diff
@@ -285,10 +285,7 @@ export class ReaderSidebar extends Component {
 		const props = {
 			path: this.props.path,
 			onClick: this.handleClick,
-			requireBackLink: false,
 			siteTitle: i18n.translate( 'Reader' ),
-			backLinkHref: this.props.returnPath || '/sites',
-			onClose: this.props.onClose && ( () => this.props.onClose() ),
```

Comment:
These were unnecessary. The sidebar only uses the latter 2 if `requireBackLink` is true. We can just remove them all.